back/engines/commercial/analysis/app/lib/analysis/q_and_a_method/base.rb

```diff
@@ -36,8 +36,13 @@ def execute(plan)
       old_answer = question.answer
       old_accuracy = question.accuracy
       old_input_ids = question.insight.inputs_ids
+      old_custom_field_ids = question.insight.custom_field_ids
       question.update!(accuracy: plan.accuracy, answer: '')
-      question.insight.update!(inputs_ids: filtered_inputs.ids)
+      custom_field_ids = {
+        main_custom_field_id: analysis.main_custom_field_id,
+        additional_custom_field_ids: analysis.additional_custom_field_ids
+      }
+      question.insight.update!(inputs_ids: filtered_inputs.ids, custom_field_ids: custom_field_ids)
```

Comment:
I don't think it should be possible to update the custom fields ids here.

Reply:
What do you mean? It's possible to update the additional fields of the analysis. So if this happened, don't we also want to use those additional fields when regenerating?